backend/src/interface/image_validation.rs

```diff
@@ -17,9 +19,42 @@ pub trait ImageValidation: Send + Sync {
 
 /// Enum describing possible ways an image validation can go wrong
 #[derive(Debug, Error)]
-pub enum ImageError {
+pub enum ImageValidationError {
     /// Error returned when an image contains invalid content.
     #[error("This image contains content that is not permitted: {0}")]
     InvalidContent(String),
-    // TODO
+    /// Error returned when the response json could not be returned.
+    #[error("The api response json could not be decoded. Image validation failed.")]
+    JsonDecodeFailed,
+    /// The response evaluation is empty.
+    #[error("The api responded with no values.")]
+    InvalidResponse,
+    /// Error returned when the api request fails.
+    #[error("The provided rest request, could not be send. Image validation failed.")]
+    RestRequestFailed,
+    /// Image could not be decoded
+    #[error("The provided image could not be decoded to base64: {0}")]
+    ImageEncodeFailed(String),
+    /// An api related error. Returns the error provided by the api.
+    #[error("The api responded with error '{0}'.")]
+    ApiResponseError(String),
+    /// Something during the token generation went wrong.
+    #[error("{0}")]
+    TokenGenerationError(#[from] google_jwt_auth::error::TokenGenerationError),
+    /// Some reqwest error occurred.
+    #[error("{0}")]
+    ReqwestError(#[from] reqwest::Error),
+    /// The json file could not be read.
+    #[error("The json file could not be read: {0}")]
+    FileReaderError(#[from] io::Error),
+}
+
+/// Structure that contains all information necessary for the image validation component.
+pub struct ImageValidationInfo {
+    /// Five numbers between 0 to 6 to set each level of a category.
+    pub acceptance: [u8; 5],
+    /// This json is needed to request authentication tokens.
+    pub json_path: String,
```

Comment:
i think it is cleaner to place thi file contents here and read the file in the config reader. Reading from files is imo ungly deendency out of the layer (which is unexpected)

Reply:
I agree, but I would not change that (yet). For that change google-jwt-auth must be changed aswell as it depends on the json string. But we should keep that in mind for the next "big" changes. Then we update both.

Reply:
Doesn't google-jwt-auth just take the file contents? it looked like you read it here